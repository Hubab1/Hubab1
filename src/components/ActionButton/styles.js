import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
