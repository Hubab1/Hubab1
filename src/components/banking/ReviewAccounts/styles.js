import { css } from 'emotion';
import styled from '@emotion/styled';

export const CardSection = styled.div`
    padding: 15px;
`

export const Card = styled.div`
    margin-bottom: 20px;
    border: 1px solid #EEEEEE;	
    border-radius: 5px;	
    box-shadow: 0 2px 4px 0 rgba(197,197,197,0.5);
    text-align: left;
`

export const totalContainer = css`
    background-color: rgba(86,186,130,0.1);
    display: flex;
    justify-content: space-between;
    padding: 12px 15px;
    width: 100%;
    position: relative;
    left: -15px;
    bottom: -15px;
`

export const incomeEntry = css`
    display: flex;    
    justify-content: space-between;
    margin: 0 0 15px 0;
`