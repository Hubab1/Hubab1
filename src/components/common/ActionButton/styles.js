import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
