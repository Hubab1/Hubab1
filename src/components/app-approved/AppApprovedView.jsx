import clsx from 'clsx';
import { Bold, H1, leftText, P, SpacedH3 } from 'assets/styles';
import approvedSign from 'assets/images/approvedSign.svg';
import Grid from '@material-ui/core/Grid';
import lightbulb from 'assets/images/lightbulb.png';
import { prettyCurrency } from 'utils/misc';
import Box from '@material-ui/core/Box';
import ActionButton from 'components/common/ActionButton/ActionButton';
import React from 'react';
import { MILESTONE_LEASE_SENT, ROLE_OCCUPANT } from 'app/constants';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from 'emotion';
import { AdverseActionNoticeButton, LINK_BUTTON } from 'components/AdverseActionNoticeButton';

export const ApprovedImage = styled.img`
    padding-top: 10px;
    height: 200px;
`;

export const applicationUnit = css`
    color: #454b57;
    font-size: 14px;
    line-height: 17px;
    text-align: center;
    padding-top: 10px;
`;

export const BulbImage = styled.img`
    width: 46px;
    height: 42px;
`;

export const securityDepositHelpText = css`
    color: #454b57;
    font-size: 14px;
    line-height: 17px;
    text-align: left;
`;

export const gridContainer = css`
    min-height: 100px;
`;

export const securityDepositTip = css`
    margin-top: 28px;
`;

export const aanDocButton = css`
    outline: none;
`;

export const AppApprovedView = ({ profile, configuration, applicant, setShowPaymentDetails }) => {
    const { unit, security_deposit: securityDeposit } = profile;

    const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
    const unitNumber = !!unit && !!unit.unit_number ? ` Unit ${unit.unit_number}` : '';

    const leaseSent = !!profile.events.find((e) => String(e.event) === String(MILESTONE_LEASE_SENT));
    const isOccupant = applicant.role === ROLE_OCCUPANT;

    let subtitle;
    if (isOccupant) {
        subtitle = `We'll let you know when everything has been finalized.`;
    } else if (leaseSent) {
        subtitle = `All that's left to do is review your payment details and sign the lease.`;
    } else {
        subtitle = `We'll send an email with instructions on how to sign the lease shortly.`;
    }

    return (
        <>
            <div className={clsx({ 'hide-element': false })}>
                <H1>{`You've Been Approved!`}</H1>
                <SpacedH3>{subtitle}</SpacedH3>
                <ApprovedImage src={approvedSign} />
                <div id="application-unit" className={applicationUnit}>
                    {buildingName}
                    {unitNumber}
                </div>
                <div className={gridContainer}>
                    {securityDeposit && (
                        <Grid
                            classes={{ root: securityDepositTip }}
                            container
                            justify={'center'}
                            className="security-deposit-container"
                            marginTop={35}
                        >
                            <Grid item xs={2}>
                                <BulbImage alt="light bulb" src={lightbulb} />
                            </Grid>
                            <Grid item xs={9} classes={{ root: leftText }}>
                                <span className={securityDepositHelpText}>
                                    A{' '}
                                    <Bold fontSize={14} fontWeight={600}>
                                        {prettyCurrency(securityDeposit)} security deposit{' '}
                                    </Bold>
                                    is required for this application.&nbsp;
                                </span>
                                <AdverseActionNoticeButton componentType={LINK_BUTTON} />
                            </Grid>
                        </Grid>
                    )}
                    {!isOccupant && leaseSent && (
                        <Box margin="28px 0 0 0">
                            <P textAlign="left" fontSize={12} color="#000000">
                                The lease linked below constitutes a legal agreement between you and Landlord. Funnel
                                Leasing does not provide legal advice, and we recommend that you consult your legal
                                counsel before accepting these terms.
                            </P>
                            <ActionButton
                                className="show-payments"
                                onClick={() => setShowPaymentDetails(true)}
                                marginTop={30}
                            >
                                Continue
                            </ActionButton>
                        </Box>
                    )}
                </div>
            </div>
        </>
    );
};

AppApprovedView.propTypes = {
    applicant: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
    profile: PropTypes.object.isRequired,
    setShowPaymentDetails: PropTypes.func.isRequired,
};
