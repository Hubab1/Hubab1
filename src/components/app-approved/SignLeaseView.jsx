import { arrowIcon, blackLinkRoot, H1, LinkButton, SpacedH3 } from 'assets/styles';
import { PaymentDetailsCard } from 'components/payment-details/PaymentDetailsCard';
import ActionButton from 'components/common/ActionButton/ActionButton';
import ArrowBackIos from '@material-ui/icons/ArrowBackIos';
import React, { useEffect } from 'react';
import hsclient from 'utils/hsclient';
import API from 'app/api';
import { APPLICATION_EVENTS, DOCUMENT_TYPE_LEASE, ROUTES } from 'app/constants';
import PropTypes from 'prop-types';

export const SignLeaseView = ({
    payables,
    profile,
    setShowPaymentDetails,
    fetchPayments,
    history,
    applicantUpdated,
}) => {
    useEffect(() => {
        fetchPayments();

        hsclient.on('sign', async () => {
            // ensure FE has the applicant signed milestone before navigating to next screen
            const newApplicant = await API.fetchApplicant();
            const leaseSignedMilestone = newApplicant.events.find(
                (e) => parseInt(e.event) === parseInt(APPLICATION_EVENTS.MILESTONE_APPLICANT_SIGNED_LEASE)
            );
            if (!leaseSignedMilestone) {
                newApplicant.events.push({
                    event: APPLICATION_EVENTS.MILESTONE_APPLICANT_SIGNED_LEASE,
                    milestone: true,
                });
            }
            applicantUpdated(newApplicant);
            // lease may not be ready by the time of navigation to the lease signed page
            setTimeout(() => history.push(ROUTES.LEASE_SIGNED), 2500);
        });
        return () => {
            hsclient.off('sign');
        };
    }, [applicantUpdated, fetchPayments, history]);

    const openEmbeddedSigning = async () => {
        const data = await API.embeddedSigningUrl(DOCUMENT_TYPE_LEASE);
        const url = data.url;
        const testMode = data.test_mode !== false;
        if (url) {
            hsclient.open(url, {
                testMode: testMode,
                skipDomainVerification: true, // TODO: update domain in HS and then change this back
                allowCancel: false,
                allowDecline: false,
            });
        }
    };

    return (
        <>
            <div>
                <H1>{`Payment Details`}</H1>
                <SpacedH3>{`Make sure everything looks good before signing the lease.`}</SpacedH3>
                <PaymentDetailsCard profile={profile} payables={payables} />
            </div>
            <ActionButton className="sign-lease" onClick={openEmbeddedSigning} marginTop={30}>
                View & Sign Lease
            </ActionButton>
            <br />
            <LinkButton className={blackLinkRoot} onClick={() => setShowPaymentDetails(false)}>
                <ArrowBackIos classes={{ root: arrowIcon }} /> Go Back
            </LinkButton>
        </>
    );
};

SignLeaseView.propTypes = {
    applicantUpdated: PropTypes.func.isRequired,
    fetchPayments: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    profile: PropTypes.object.isRequired,
    payables: PropTypes.array.isRequired,
    setShowPaymentDetails: PropTypes.func.isRequired,
};
