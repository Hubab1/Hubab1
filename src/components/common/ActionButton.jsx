import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}