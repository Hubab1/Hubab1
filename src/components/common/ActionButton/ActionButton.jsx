import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}