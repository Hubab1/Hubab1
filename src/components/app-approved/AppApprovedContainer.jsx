import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { applicantUpdated } from 'reducers/applicant';
import { ROUTES } from 'app/constants';

import captureRoute from 'app/captureRoute';
import { fetchPayments } from 'reducers/payments';
import { SignLeaseView } from 'components/app-approved/SignLeaseView';
import { AppApprovedView } from 'components/app-approved/AppApprovedView';

export const AppApprovedContainer = ({ profile, configuration, history, applicantUpdated, applicant, payables }) => {
    const [showPaymentDetails, setShowPaymentDetails] = useState(false);

    if (!profile || !configuration) return null;

    if (showPaymentDetails) {
        return (
            <SignLeaseView
                applicantUpdated={applicantUpdated}
                fetchPayments={fetchPayments}
                history={history}
                profile={profile}
                payables={payables}
                setShowPaymentDetails={setShowPaymentDetails}
            />
        );
    }

    return (
        <AppApprovedView
            applicant={applicant}
            configuration={configuration}
            profile={profile}
            setShowPaymentDetails={setShowPaymentDetails}
        />
    );
};

AppApprovedContainer.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
    updateApplicant: PropTypes.object,
    applicant: PropTypes.object,
    history: PropTypes.object,
    applicantUpdated: PropTypes.func,
    payables: PropTypes.array,
};

const mapStateToProps = (state) => ({
    profile: state.renterProfile,
    applicant: state.applicant,
    configuration: state.configuration,
    payables: state.payables || [],
});

const mapDispatchToProps = {
    applicantUpdated,
    fetchPayments,
};

export default connect(mapStateToProps, mapDispatchToProps)(captureRoute(AppApprovedContainer, ROUTES.APP_APPROVED));
