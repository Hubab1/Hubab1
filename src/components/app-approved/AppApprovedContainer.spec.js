import { shallow } from 'enzyme';
import { AppApprovedContainer } from 'components/app-approved/AppApprovedContainer';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { PaymentDetailsCard } from 'components/payment-details/PaymentDetailsCard';
import React from 'react';
import { AppApprovedView } from 'components/app-approved/AppApprovedView';
import { SignLeaseView } from 'components/app-approved/SignLeaseView';
import { LinkButton } from 'assets/styles';

const buildProps = (buildingName = 'Fake Building', streetAddress = '123 Fake Street', unitNumber = '2B') => {
    return {
        profile: {
            events: [{ event: '270' }],
            unit: {
                unit_number: unitNumber,
            },
            last_status_change: {
                created_at: '2020-05-06 04:06:23',
            },
        },
        configuration: {
            community: {
                building_name: buildingName,
                normalized_street_address: streetAddress,
            },
        },
        history: {
            push: jest.fn(),
        },
        applicant: {
            client: {
                person: {
                    name: 'John Doe',
                },
            },
        },
        applicantUpdated: jest.fn(),
        fetchPayments: jest.fn(),
        payables: [],
    };
};

describe('it displays the right section', () => {
    const props = buildProps('Fake Building', '123 Fake Street', null);
    const wrapper = shallow(<AppApprovedContainer {...props} />);

    it('shows payment details section when clicking continue', () => {
        expect(wrapper.find(AppApprovedView)).toHaveLength(1);
        wrapper.dive().find(ActionButton).simulate('click');
        expect(wrapper.find(PaymentDetailsCard)).toBeTruthy();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('hides payment details section when clicking back', () => {
        expect(wrapper.find(SignLeaseView)).toHaveLength(1);
        wrapper.dive().find(LinkButton).simulate('click');
        expect(wrapper.find(PaymentDetailsCard)).toHaveLength(0);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
