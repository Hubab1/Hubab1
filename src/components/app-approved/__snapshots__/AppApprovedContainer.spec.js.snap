// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays the right section hides payment details section when clicking back 1`] = `
<AppApprovedView
  applicant={
    Object {
      "client": Object {
        "person": Object {
          "name": "John Doe",
        },
      },
    }
  }
  configuration={
    Object {
      "community": Object {
        "building_name": "Fake Building",
        "normalized_street_address": "123 Fake Street",
      },
    }
  }
  profile={
    Object {
      "events": Array [
        Object {
          "event": "270",
        },
      ],
      "last_status_change": Object {
        "created_at": "2020-05-06 04:06:23",
      },
      "unit": Object {
        "unit_number": null,
      },
    }
  }
  setShowPaymentDetails={[Function]}
/>
`;

exports[`it displays the right section shows payment details section when clicking continue 1`] = `
<SignLeaseView
  applicantUpdated={[MockFunction]}
  fetchPayments={[Function]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  payables={Array []}
  profile={
    Object {
      "events": Array [
        Object {
          "event": "270",
        },
      ],
      "last_status_change": Object {
        "created_at": "2020-05-06 04:06:23",
      },
      "unit": Object {
        "unit_number": null,
      },
    }
  }
  setShowPaymentDetails={[Function]}
/>
`;
